### ISSUE TITLE
[BUG] Failed to check ruleset; file not found

### ISSUE THREAD

**Issue by noot-to-the-noot** [terraform-linters/tflint#1465]
## Introduction

I hope this issue belongs to tflint directly and is not a problem related to the [ruleset](https://github.com/terraform-linters/tflint-ruleset-aws) we're using.

Currently we're experiencing a problem while using tflint, that doesn't have a apparent reason.
While running `tflint --config=${TFLINT_PLUGIN_CONFIG_FILE}` where the path comes from environment via `TFLINT_PLUGIN_CONFIG_FILE=/etc/tflint/.tflint.hcl`  with the current config: 
```
plugin "aws" {
  enabled = true
  version = "0.13.3"
  source  = "github.com/terraform-linters/tflint-ruleset-aws"
  deep_check = true
}

config {
  module = true
  force = false
  disabled_by_default = false
}

rule "aws_acm_certificate_invalid_private_key" {
  enabled = false
}
```

It only gives out the error `Failed to check ruleset; file not found`.

## Expected Behavior

After running tflint prints errors inside the terraform code if any are present.

## Actual behavior

After running tflint the error message: `Failed to check ruleset; file not found` gets logged and the program exits with a non zero exit code.

## Step to Reproduce

I can't really tell how to reproduce since I can't tell what currently is going wrong or why the program crashes.
I could only narrow the error down to the function that crashes ([terraform_workspace_remote.go](https://github.com/terraform-linters/tflint/blob/master/rules/terraformrules/terraform_workspace_remote.go#L44))

Here is a shortened and censored trace log. Unfortunately I cannot provide more.

```
tflint --config=/etc/tflint/.tflint.hcl --loglevel=trace
12:59:24 config.go:115: [INFO] Load config: /etc/tflint/.tflint.hcl
12:59:24 config.go:242: [DEBUG] Config loaded
12:59:24 config.go:243: [DEBUG]   Module: true
12:59:24 config.go:244: [DEBUG]   Force: false
12:59:24 config.go:245: [DEBUG]   IgnoreModules:
12:59:24 config.go:249: [DEBUG]   Varfiles: 
12:59:24 config.go:250: [DEBUG]   Variables: 
12:59:24 config.go:251: [DEBUG]   DisabledByDefault: false
12:59:24 config.go:252: [DEBUG]   PluginDir: 
12:59:24 config.go:253: [DEBUG]   Format: 
12:59:24 config.go:254: [DEBUG]   Rules:
12:59:24 config.go:256: [DEBUG]     aws_acm_certificate_invalid_private_key: false
12:59:24 config.go:258: [DEBUG]   Plugins:
12:59:24 config.go:260: [DEBUG]     aws: enabled=true, version=0.13.3, source=github.com/terraform-linters/tflint-ruleset-aws
12:59:24 option.go:49: [DEBUG] CLI Options
12:59:24 option.go:50: [DEBUG]   Module: false
12:59:24 option.go:51: [DEBUG]   Force: false
12:59:24 option.go:52: [DEBUG]   IgnoreModules:
12:59:24 option.go:56: [DEBUG]   EnableRules: 
12:59:24 option.go:57: [DEBUG]   DisableRules: 
12:59:24 option.go:58: [DEBUG]   Only: 
12:59:24 option.go:59: [DEBUG]   EnablePlugins: 
12:59:24 option.go:60: [DEBUG]   Varfiles: 
12:59:24 option.go:61: [DEBUG]   Variables: 
12:59:24 option.go:62: [DEBUG]   Format: 
12:59:24 loader.go:53: [INFO] Initialize new loader
12:59:24 loader.go:64: [INFO] Module manifest file found. Initializing...
12:59:24 loader.go:250: [DEBUG] Parsing the module manifest file: <censored>
12:59:24 loader.go:88: [INFO] Module inspection is enabled. Building a root module with children...
12:59:24 loader.go:232: [DEBUG] Trying to load the module: <censored>
12:59:24 loader.go:232: [DEBUG] Trying to load the module: <censored>
12:59:24 loader.go:232: [DEBUG] Trying to load the module: <censored>
12:59:24 loader.go:232: [DEBUG] Trying to load the module: <censored>
12:59:24 loader.go:232: [DEBUG] Trying to load the module: <censored>
12:59:24 loader.go:232: [DEBUG] Trying to load the module: <censored>
12:59:24 loader.go:232: [DEBUG] Trying to load the module: <censored>
12:59:24 loader.go:232: [DEBUG] Trying to load the module: <censored>
12:59:24 loader.go:232: [DEBUG] Trying to load the module: <censored>
12:59:24 loader.go:232: [DEBUG] Trying to load the module: <censored>
12:59:24 loader.go:232: [DEBUG] Trying to load the module: <censored>
12:59:24 loader.go:232: [DEBUG] Trying to load the module: <censored>
12:59:24 loader.go:232: [DEBUG] Trying to load the module: <censored>
12:59:24 loader.go:232: [DEBUG] Trying to load the module: <censored>
12:59:24 loader.go:232: [DEBUG] Trying to load the module: <censored>
12:59:24 loader.go:232: [DEBUG] Trying to load the module: <censored>
12:59:24 loader.go:232: [DEBUG] Trying to load the module: <censored>
12:59:24 loader.go:131: [INFO] Load values files
12:59:24 runner.go:50: [INFO] Initialize new runner for root
12:59:24 runner.go:174: [DEBUG] `default_tags` has been marked as unknown
12:59:24 runner.go:50: [INFO] Initialize new runner for module.<censored>
12:59:24 runner.go:174: [DEBUG] `default_tags` has been marked as unknown
12:59:24 runner.go:50: [INFO] Initialize new runner for module.<censored>
12:59:24 runner.go:174: [DEBUG] `subnets` has been marked as unknown
12:59:24 runner.go:174: [DEBUG] `vpc_id` has been marked as unknown
12:59:24 runner.go:174: [DEBUG] `default_tags` has been marked as unknown
12:59:24 runner.go:50: [INFO] Initialize new runner for module.<censored>
12:59:24 runner.go:174: [DEBUG] `default_tags` has been marked as unknown
12:59:24 runner.go:50: [INFO] Initialize new runner for module.<censored>
12:59:24 runner.go:174: [DEBUG] `vpc_id` has been marked as unknown
12:59:24 runner.go:174: [DEBUG] `rules` has been marked as unknown
12:59:24 runner.go:174: [DEBUG] `tags` has been marked as unknown
12:59:24 runner.go:50: [INFO] Initialize new runner for module.<censored>
12:59:24 runner.go:174: [DEBUG] `tags` has been marked as unknown
12:59:24 runner.go:174: [DEBUG] `vpc_id` has been marked as unknown
12:59:24 runner.go:174: [DEBUG] `rules` has been marked as unknown
12:59:24 runner.go:50: [INFO] Initialize new runner for module.<censored>
12:59:24 runner.go:174: [DEBUG] `vpc_id` has been marked as unknown
12:59:24 runner.go:174: [DEBUG] `rules` has been marked as unknown
12:59:24 runner.go:174: [DEBUG] `tags` has been marked as unknown
12:59:24 runner.go:50: [INFO] Initialize new runner for module.<censored>
12:59:24 runner.go:174: [DEBUG] `default_tags` has been marked as unknown
12:59:24 runner.go:174: [DEBUG] `user_name` has been marked as unknown
12:59:24 runner.go:174: [DEBUG] `path` has been marked as unknown
12:59:24 runner.go:50: [INFO] Initialize new runner for module.<censored>
12:59:24 runner_eval.go:41: [INFO] unevaluable expression found in <censored>. TFLint ignores unevaluable expressions.
12:59:24 runner_eval.go:41: [INFO] unevaluable expression found in <censored>. TFLint ignores unevaluable expressions.
12:59:24 runner_eval.go:41: [INFO] unevaluable expression found in <censored>. TFLint ignores unevaluable expressions.
12:59:24 runner_eval.go:41: [INFO] unevaluable expression found in <censored>. TFLint ignores unevaluable expressions.
12:59:24 discovery.go:79: [DEBUG] Find plugin path: /root/.tflint.d/plugins/github.com/terraform-linters/tflint-ruleset-aws/0.13.3/tflint-ruleset-aws
12:59:24 discovery.go:43: [INFO] Plugin `aws` found
12:59:24 provider.go:64: [INFO] Prepare rules
12:59:24 provider.go:92: [INFO]   5 default rules enabled
12:59:24 terraform_deprecated_interpolation.go:48: [TRACE] Check `terraform_deprecated_interpolation` rule for `root` runner
12:59:24 terraform_empty_list_equality.go:46: [TRACE] Check `terraform_empty_list_equality` rule for `root` runner
12:59:24 terraform_module_pinned_source.go:61: [TRACE] Check `terraform_module_pinned_source` rule for `root` runner
12:59:24 terraform_module_pinned_source.go:89: [DEBUG] Walk `access_keys_rotation.source` attribute
12:59:24 terraform_module_pinned_source.go:89: [DEBUG] Walk `<censored>.source` attribute
12:59:24 terraform_module_pinned_source.go:89: [DEBUG] Walk `<censored>.source` attribute
12:59:24 terraform_module_pinned_source.go:89: [DEBUG] Walk `<censored>.source` attribute
12:59:24 terraform_module_pinned_source.go:89: [DEBUG] Walk `<censored>.source` attribute
12:59:24 terraform_module_pinned_source.go:89: [DEBUG] Walk `<censored>.source` attribute
12:59:24 terraform_module_pinned_source.go:89: [DEBUG] Walk `<censored>.source` attribute
12:59:24 terraform_module_pinned_source.go:89: [DEBUG] Walk `<censored>.source` attribute
12:59:24 terraform_module_pinned_source.go:89: [DEBUG] Walk `<censored>.source` attribute
12:59:24 terraform_module_version.go:58: [TRACE] Check `terraform_module_version` rule for `root` runner
12:59:24 terraform_module_version.go:85: [DEBUG] Walk `<censored>.source` attribute
12:59:24 terraform_module_version.go:85: [DEBUG] Walk `<censored>.source` attribute
12:59:24 terraform_module_version.go:85: [DEBUG] Walk `<censored>.source` attribute
12:59:24 terraform_module_version.go:85: [DEBUG] Walk `<censored>.source` attribute
12:59:24 terraform_module_version.go:85: [DEBUG] Walk `<censored>.source` attribute
12:59:24 terraform_module_version.go:85: [DEBUG] Walk `<censored>.source` attribute
12:59:24 terraform_module_version.go:85: [DEBUG] Walk `<censored>.source` attribute
12:59:24 terraform_module_version.go:85: [DEBUG] Walk `<censored>.source` attribute
12:59:24 terraform_module_version.go:85: [DEBUG] Walk `<censored>.source` attribute
12:59:24 terraform_workspace_remote.go:50: [TRACE] Check `terraform_workspace_remote` rule for `root` runner
Failed to check ruleset; file not found
```

## Additional Context

```console
$ tflint -v
TFLint version 0.39.1
$ terraform -v
Terraform v1.2.6
on linux_amd64
+ provider registry.terraform.io/hashicorp/archive v2.2.0
+ provider registry.terraform.io/hashicorp/aws v4.24.0
+ provider registry.terraform.io/hashicorp/local v2.2.3

**Comment by bendrucker** [comment]
Nothing above points to the issue. We really need a reproduction, including Terraform config, to diagnose issues. Given that an empty config doesn't reproduce this issue you could start cutting out bits of your config until you identify a minimal config that causes the error that you can scrub/share.

**Comment by syphernl** [comment]
Related ticket: https://github.com/terraform-linters/tflint-bundle/issues/37

**Comment by noot-to-the-noot** [comment]
@bendrucker I'll try to work out a minimal setup that reproduces the error, give me some time, since I'll have to get approvement first.

**Comment by noot-to-the-noot** [comment]
Okay so here is a minimal setup that produces the error on my system, hope you can reproduce it aswell
[setup.zip](https://github.com/terraform-linters/tflint/files/9250023/setup.zip)

**Comment by danielroettig** [comment]
I noticed the same issue after upgrading from v0.38.1 to v0.39.1. When trying to come up with a minimal terraform setup to reproduce the issue, it seemed the issue starts to appear as soon as I try to reference a module from a subdirectory, e.g.:

```
module "fails" {
  source     = "./somemodule/"
}
```

**Comment by bendrucker** [comment]
Excellent, in your archive I can reproduce via the following:

```sh
cd simple_setup
# copy config
pbpaste > .tflint.hcl
terraform init
tflint --init
tflint
```

From here we can capture the error in a debugger and figure out exactly where it originates. Seems like this is coming from the AWS plugin based on the fact that disabling it eliminates the error.
